### PR TITLE
Add kotlin code block for chapter array and linkedlist

### PR DESCRIPTION
**Pull request by curtishd**
If this pull request (PR) pertains to **Chinese-to-English translation**, please confirm that you have read the contribution guidelines and complete the checklist below:

- [ ] This PR represents the translation of a single, complete document, or contains only bug fixes.
- [ ] The translation accurately conveys the original meaning and intent of the Chinese version. If deviations exist, I have provided explanatory comments to clarify the reasons.

If this pull request (PR) is associated with **coding or code transpilation**, please attach the relevant console outputs to the PR and complete the following checklist:

- [x] I have thoroughly reviewed the code, focusing on its formatting, comments, indentation, and file headers.
- [x] I have confirmed that the code execution outputs are consistent with those produced by the reference code (Python or Java).
- [x] The code is designed to be compatible on standard operating systems, including Windows, macOS, and Ubuntu.

### CODE REVIEW

**Comment by curtishd** [comment]
Looks like someone has already submitted array.md in this chapter. So, I will submit the rest.

File: docs/chapter_array_and_linkedlist/linked_list.md [hunk 1]
```diff
@@ -162,7 +162,13 @@
 === "Kotlin"
 
     ```kotlin title=""
-
+    /* 链表节点类 */
+    class ListNode(
+        x: Int // 主构造函数
+    ){
```
Comment (krahets) [inline docs/chapter_array_and_linkedlist/linked_list.md:168]: ) {

Please FORMAT all the code before filling them into documents.

File: docs/chapter_array_and_linkedlist/linked_list.md [hunk 2]
```diff
@@ -382,6 +388,18 @@
...
+    val n0 = ListNode(1)
+    val n1 = ListNode(3)
+    val n2 = ListNode(2)
+    val n3 = ListNode(5)
+    val n4 = ListNode(4)
+    // 构建节点之间的引用
+    n0.next = n1;
+    n1.next = n2;
+    n2.next = n3;
+    n3.next = n4;
```
Comment (krahets) [inline docs/chapter_array_and_linkedlist/linked_list.md:403]: Remove the redundant empty line

**Comment by krahets** [comment]
> Looks like someone has already submitted array.md in this chapter. So, I will submit the rest.

You can link PR #1156 to make it clear.

File: docs/chapter_array_and_linkedlist/linked_list.md [hunk 3]
```diff
@@ -162,7 +162,13 @@
 === "Kotlin"
 
     ```kotlin title=""
-
+    /* 链表节点类 */
+    class ListNode(
+        x: Int // 主构造函数
```
Comment (krahets) [inline docs/chapter_array_and_linkedlist/linked_list.md:167]: `主构造函数` -> `构造方法`

File: docs/chapter_array_and_linkedlist/linked_list.md [hunk 4]
```diff
@@ -643,7 +661,14 @@
 === "Kotlin"
 
     ```kotlin title=""
-
+    /* 双向链表节点类 */
+    class ListNode(
+        x: Int // 主构造函数
```
Comment (krahets) [inline docs/chapter_array_and_linkedlist/linked_list.md:666]: Same as above. Please check other similar comments.

**Review by krahets — APPROVED** [review]
Thanks for the code!

**Pull request merged**